tests/helpers.nim

```diff
@@ -117,3 +117,10 @@ proc checkExpiringInternal(cond: proc(): bool {.raises: [Defect], gcsafe.} ): Fu
 
 template checkExpiring*(code: untyped): untyped =
   check await checkExpiringInternal(proc(): bool = code)
+
+type
+  Awaiter* = ref object of RootObj
```

Comment:
Why this wrapper?

Reply:
The trick to use a global inside a gcsafe.

Reply:
That seems very weird, `Future` is already a `ref object of RootObj`, so this shouldn't change anything, and you are not even using it as a global